src/content/en/updates/posts/2016/03/web-enabled-internet-of-things.markdown

```diff
@@ -0,0 +1,545 @@
+App (aka PWA), which means that the site has a responsive app like
+user experience, works offline and can be added to the home screen.
+Google has a really good explainer about PWAs, which I urge you to
+check out:
+
+[Progressive Web Apps](https://developers.google.com/web/progressive-web-apps)
+
+As a proof of concept, I have been building a small device using
+the Intel&reg; Edison Arduino breakout board. It contains a temperature
+sensor (TMP36) as well as an actuator (colored LED cathode). The
+schematics can be found at the end of this article.
+
+![Breadboard](/web/updates/images/2016/03/web-enabled-internet-of-things/breadboard.jpg)
+
+The Edison is an interesting device as it can run a full Linux
+distribution and therefore allows me to program it easily using
+Node.js. The installer allows for installing the Intel&reg; XDK which
+makes it easy to get started, though you can program and upload
+to your device manually as well.
+
+> NOTE: It is possible to use Brillo or Ostro instead of the
+> default OS software, in which case you should follow the documentation
+> for these OSes for now to get a Node.js application running on the
+> device.
+
+For my Node.js app, I required three node modules, as well as their
+dependencies: `eddystone-beacon`, `parse-color`, `johnny-five`. The
+former will automatically install `noble` which is the node module
+that I use for talking Bluetooth Low Energy. 
+
+> NOTE: It is important to not list `noble` as a dependency in the
+> `package.json` file, as you need to use the same noble instance as
+> `eddystone-beacon`, in order for them to work together.
+
+You can find more info [here](https://github.com/don/node-eddystone-beacon/issues/30)
+
+The `package.json` file for the project looks like this:
+
+{% highlight javascript %}
+{
+  "name": "edison-webbluetooth-demo-server",
+  "version": "1.0.0",
+  "main": "main.js",
+  "engines": {
+    "node": ">=0.10.0"
+  },
+  "dependencies": {
+    "eddystone-beacon" : "1.0.5",
+    "parse-color": "1.0.0",
+    "johnny-five": "0.9.30"
+  }
+}
+{% endhighlight %}
+
+## Announcing the web site
+
+The beta version of Chrome on Android supports Physical Web, which
+allows Chrome to see URLs being broadcasted by devices around it.
+There are a few requirements, such as the sites need to be publicly
+accessible and use HTTPS.
+
+My demo app is hosted at <https://edison-webbt.appspot.com/> but the
+Eddystone protocol has a limit on the size of the URL (max 18 bytes,
+more info
+[here](https://github.com/google/eddystone/blob/master/eddystone-url/docs/config-service-spec.md#34-uri-data)), 
+so I have to use a URL shortener in order for it to work.
+
+Broadcasting the URL is quite simple. Import the required libraries and
+call a few functions. One way of doing it is calling `advertiseUrl`, when
+the BLE chip is turned on:
+
+{% highlight javascript %}
+var beacon = require("eddystone-beacon");
+var bleno = require('eddystone-beacon/node_modules/bleno');
```

Comment:
No, because then it wont work with eddystone-beacon as documented elsewhere in this article